src/games/models/game.ts

```diff
@@ -78,6 +89,41 @@ export class Game extends MongooseDocument {
   @Prop({ unique: true, sparse: true })
   logSecret?: string;
 
+  async serialize(): Promise<GameDto> {
+    const playersService = app.get(PlayersService);
+    const gameServersService = app.get(GameServersService);
+
+    return {
+      id: this.id,
+      launchedAt: this.launchedAt,
+      endedAt: this.endedAt,
+      number: this.number,
+      slots: await Promise.all(
+        this.slots.map(async (slot) => ({
+          player: await playersService.getById(slot.player),
+          team: slot.team,
+          gameClass: slot.gameClass,
+          status: slot.status,
+          connectionStatus: slot.connectionStatus,
+        })),
+      ),
+      map: this.map,
+      state: this.state,
+      connectInfoVersion: this.connectInfoVersion,
+      stvConnectString: this.stvConnectString,
+      logsUrl: this.logsUrl,
+      demoUrl: this.demoUrl,
+      error: this.error,
+      gameServer: this.gameServer
+        ? await gameServersService.getById(this.gameServer)
+        : undefined,
```

Comment:
doesn't this return undefined if the id isn't there?

Reply:
It will throw an error.